helpers/telemetry.py

```diff
@@ -66,31 +62,26 @@ def __init__(
         self.commit_id = commit_id
         self.commit_sha = commit_sha
         self.owner_id = owner_id
-        self.repo_slug = None
-        self.owner_slug = None
-        self.commit_slug = None
         self.populated = False
 
     def populate(self):
         if self.populated:
             return
 
         repo = None
-        owner = None
         commit = None
         dbsession = get_db_session()
 
         if self.repo_id:
-            repo = (
-                dbsession.query(Repository)
-                .filter(Repository.repoid == self.repo_id)
-                .first()
-            )
-            owner = repo.owner
             if not self.owner_id:
-                self.owner_id = owner.ownerid
+                repo = (
+                    dbsession.query(Repository)
+                    .filter(Repository.repoid == self.repo_id)
+                    .first()
+                )
+                self.owner_id = repo.ownerid
 
-            if self.commit_sha:
+            if self.commit_sha and not self.commit_id:
```

Comment:
Could you have a sha but not an id?

Reply:
i think so, don't remember where i saw this. i think it was maybe like, a webhook or GH API response or something that didn't include our DB id but did include the sha